### PR TITLE
#438 #437 Added home page similar to the one presented in Mapstore.

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
Changes introduced by PR:
- Added custom "Home" plugin and activated it for map, geostory, dashboard editors. Newly created context will also have "Home" plugin active by default.
- Created new "Contexts" plugin that is a mix of two existing plugins: "ContextManager" and "Maps" (or "Dashboards", "Geostories"). This plugin allows managing of contexts from the home page at a separate content tab.
- Created custom components "ResourceCard" and "ResourceGrid" with support of customizing toolbar based on resource type. Right now it's possible to customize behavior of resource editing tool, by running custom action per resource type. It's also possible to either show or hide button that allows editing based on resource type. "FeaturedResourcesGrid" is utilizing customized "ResourceGrid" component to provide different tools set for featured Contexts.
- Minor fixes to the home page layout

Part of the changes like "FeaturedResourcesGrid" and "ResourceGrid" eventually will need to be backported to the Mapstore upstream.

### CODE REVIEW

**Comment by tdipisa** [comment]
@offtherailz can you please review this?

**Review by offtherailz — CHANGES_REQUESTED** [review]
## Review

Looking at this PR and testing everything looks good. ;Merging this solution may help us to test the final result on the environment and evaluate things, so I should merge it @tdipisa 

Only things that i noticed:

- [ ] The "Contexts" plugin, if possible, should not load on search if not present ( Actually it loads anyway on search because of the epic)


## Port to MapStore 

Anyway many of the changes are pure duplications, due to limits of home page plugins to be customized (they are not used anywhere else than in standard MapStore versions). This help to update our current plugins and to maintain them better, avoiding any duplication. 

here I list the things that can be included in MapStore @alexander-fedorenko please correct me if I'm wrong
- [ ] Home plugin may be ported on MapStore, only need to customize a little bit the path to trigger (`/home` vs `/`) and maybe some behaviors for the "context" case..  For the current implementation this is the harder thing to port (even if the things to change looks simple), and we can evaluate to avoid it for the moment.
- [ ] "Contexts"  plugin can be included in mapstore as it is, adding its missing translations. 
- [ ] Custom  "ResourceCard" and "ResourceGrid", "FeaturedResourcesGrid"   components customization can be ported in MapStore.

These porting require a little of work and test. I can suggest to merge this PR first (after fixing review things) and then proceed to porting, so we can test and identify other customizations or issues that we did not identified in a first review. A final PR will remove all duplication and keep customization, that will be hopefully only configurations.

File: js/pages/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
```
[the remark this answers is not in the document]
Comment (alexander-fedorenko) [reply js/pages/Home.jsx]: Nope, changed it now to use Maps.jsx for sake of backwards compatibility.

**Comment by offtherailz** [comment]
I backported the PR on MapStore, so @alexander-fedorenko you can clean up this PR  and finalize it.

**Comment by alexander-fedorenko** [comment]
PR is updated to be in align with the changes backported to the Mapstore.
@offtherailz please re-review.

File: configs/localConfig.json [hunk 2]
```diff
@@ -725,52 +729,248 @@
...
+                "url": "https://gs-stable.geo-solutions.it/geoserver/wms",
+                "type": "wms",
+                "title": "GeoSolutions GeoServer WMS"
+              },
+              "GeoSolutions GeoServer WMTS": {
+                "url": "https://gs-stable.geo-solutions.it/geoserver/gwc/service/wmts",
+                "type": "wmts",
+                "title": "GeoSolutions GeoServer WMTS"
+              }
+            },
```
Comment (offtherailz) [inline configs/localConfig.json:825]: Here we should leave it empty or use the services configured in geOrchestra. @tdipisa what services the dashboard should have as default in geOrchestra default installation? local geoserver (csw extension is not present, so we may try to use wfs or so...)

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] We need to configure the catalogs
- [ ] I wasn't able to test it because the sub-module is not updated to the correct commit. (Please test it before)
- [ ] @tdipisa They said that home page is optional. This means that the button that links to the home from the main map should not be included by default, isn't it? 
- [ ] Home plugin should be added also to `pluginsConfig.json`

File: configs/localConfig.json (hunk 2 above)
Comment (tdipisa) [reply configs/localConfig.json:825]: > Here we should leave it empty or use the services configured in geOrchestra. @tdipisa what services the dashboard should have as default in geOrchestra default installation? local geoserver (csw extension is not present, so we may try to use wfs or so...)

@alexander-fedorenko @offtherailz please use the same have been always used until now for the [default context](https://github.com/georchestra/mapstore2-georchestra/blob/master/configs/localConfig.json#L111)

Comment (tdipisa) [reply configs/localConfig.json:825]: > @tdipisa They said that home page is optional. This means that the button that links to the home from the main map should not be included by default, isn't it?

It is correct

Comment (tdipisa) [reply configs/localConfig.json:825]: @alexander-fedorenko please proceed applying requested changes.

Comment (landryb) [reply configs/localConfig.json:825]: > > @tdipisa They said that home page is optional. This means that the button that links to the home from the main map should not be included by default, isn't it?
> 
> Of course, it is correct.

chiming in, anyway its just a matter of adding the 'Home' plugin to the context/default config, right ?

**Review by offtherailz — APPROVED** [review]
I clean up the catalogs list in dashboard, using only local geoserver as default

**Comment by ElenaGallo** [comment]
@alexander-fedorenko @tdipisa After testing the home page I found these problems:

**1_**Embed tab on Share panel is not visible if the Context is on Featured

**How to Reproduce**
See the gif below:
embed_tab


*Expected Result*
The embed tab is present if the context is on Featured


**2_**It is not possible to save a new geostory 

**How to Reproduce**
- Open MapStoreHome 
- Click on New GeoStory
- Click on Options
- Click on Save as
- Enter a title
- Click on Save

*Expected Result*
It is possible to create a new geostory

**3_** There is no context icon on the context present on Featured

home-ge

**Comment by tdipisa** [comment]
@alexander-fedorenko can you please check the above? Thank you so much @ElenaGallo.

**Comment by alexander-fedorenko** [comment]
Bugfix for the 1_ item is awaiting for review here: https://github.com/geosolutions-it/MapStore2/pull/7746
2_ Is a backend-related issue as discussed, should be resolved not in scope of this issue/PR.
3_ Require new icon as discussed with @tdipisa.

**Comment by ElenaGallo** [comment]
@tdipisa test for points 1 and 3 passed.

**Pull request merged**